src/main/java/sirius/search/RobustQueryParser.java

```diff
@@ -51,24 +51,38 @@
      * @param autoexpand   should single token queries be auto expanded. That will put a "*" after the resulting token
      *                     but limits the number of expansions to the top 256 terms.
      */
-    RobustQueryParser(String defaultField,
-                      String input,
-                      Function<String, Iterable<List<String>>> tokenizer,
-                      boolean autoexpand) {
+    public RobustQueryParser(String defaultField,
+                             String input,
+                             Function<String, Iterable<List<String>>> tokenizer,
+                             boolean autoexpand) {
         this.defaultField = defaultField;
         this.input = input;
         this.tokenizer = tokenizer;
         this.autoexpand = autoexpand;
     }
 
     /**
-     * Compiles an applies the query to the given one.
+     * Compiles and applies the query to the given one.
      *
      * @param query               the query to enhance with the parsed result
-     * @param failForEmptyQueries determines if a query was forced, see {@link Query#forceQuery(String, String,
-     *                            Function)}
+     * @param failForEmptyQueries determines if a query was forced, see {@link Query#forceQuery(String, String, *                            Function)}
      */
     void compileAndApply(Query<?> query, boolean failForEmptyQueries) {
+        QueryBuilder main = compile();
+        if (main != null) {
+            if (Index.LOG.isFINE()) {
+                Index.LOG.FINE("Compiled '%s' into '%s'", query, main);
+            }
+            query.where(Wrapper.on(main));
+        } else if (failForEmptyQueries) {
+            query.fail();
+        }
+    }
+
+    /**
+     * Compiles the query.
+     */
+    public QueryBuilder compile() {
```

Comment:
Why not simply add a method "asConstraint()" to RobustQueryParser which returns Wrapper.on(compile())